src/hotspot/share/oops/arrayOop.hpp

```diff
@@ -68,10 +68,10 @@ class arrayOopDesc : public oopDesc {
   // The header is considered the oop part of this type plus the length.
   // This is not equivalent to sizeof(arrayOopDesc) which should not appear in the code.
   static int header_size_in_bytes() {
-    size_t hs = length_offset_in_bytes() + sizeof(int);
+    int hs = length_offset_in_bytes() + (int)sizeof(int);
 #ifdef ASSERT
     // make sure it isn't called before UseCompressedOops is initialized.
-    static size_t arrayoopdesc_hs = 0;
+    static int arrayoopdesc_hs = 0;
```

Comment:
Why not just do a `checked_cast<int>` on the return statement? (or even a range assert and a static cast?)

Reply:
Also I think the current checked_cast still has problems with signed <--> unsigned.

Another technique to use an even more restricted type.  For example, if these offset_in_bytes() functions returned uint16_t, then the value can be widened to either `int` or `size_t` without a cast.

Reply:
Yes that would be a change that we could make that would help, and these offset_in_bytes functions should return an unsigned type since they're never negative.